api/task_manager_test.cc

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <seastar/core/coroutine.hh>
+
+#include "task_manager_test.hh"
+#include "api/api-doc/task_manager_test.json.hh"
+#include "task_manager/test_task.hh"
+
+
+namespace api {
+
+namespace tmt = httpd::task_manager_test_json;
+using namespace json;
+
+void set_task_manager_test(http_context& ctx, routes& r) {
+    tmt::register_test_module.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        co_await ctx.tm.invoke_on_all([] (task_manager& tm) {
+            tm.make_test_module();
+        });
+        co_return json_void();
+    });
+
+    tmt::unregister_test_module.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        co_await ctx.tm.invoke_on_all([name = "test"] (task_manager& tm) -> future<> {
+            auto& modules = tm.get_modules();
+            auto it = modules.find(name);
+            if (it != modules.end()) {
+                co_await it->second->stop();
+            }
+        });
+        co_return json_void();
+    });
+
+    tmt::register_test_task.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        sharded<task_manager>& tms = ctx.tm;
+        auto it = req->query_parameters.find("task_id");
+        task_id id = it != req->query_parameters.end() ? task_id{utils::UUID{it->second}} : task_id::create_random_id();
+        it = req->query_parameters.find("keyspace");
+        std::string keyspace = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("table");
+        std::string table = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("type");
+        std::string type = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("entity");
+        std::string entity = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("sequence_number");
+        int sequence_number = it != req->query_parameters.end() ? boost::lexical_cast<int>(it->second) : 0;
+        it = req->query_parameters.find("parent_id");
+        task_id parent_id = it != req->query_parameters.end() ? task_id{utils::UUID{it->second}} : task_id::create_null_id();
+        co_await tms.invoke_on(0, [id, name = "test", sequence_number, keyspace = std::move(keyspace), table = std::move(table), type = std::move(type), entity = std::move(entity), parent_id] (task_manager& tm) -> future<> {
```

Comment:
please define `auto module_name = "test";` as a regular statement rather than a capture.

Reply:
Alternatively define it as a static method of the module so each module can return its name.
and call it here (and elsewhere you need to refer to the module name)